### PR TITLE
Migrate/zh/cli

### PR DESCRIPTION
**Pull request by mzxyz**
Migrate all the updates for cli from latest plugBlockchain

### CODE REVIEW

File: cli/build.rs [hunk 1]
```diff
@@ -49,5 +51,5 @@ fn build_completion(shell: &Shell) {
 
 	fs::create_dir(&path).ok();
 
-	CoreParams::<NoCustom, NoCustom>::clap().gen_completions("cennznet-node", *shell, &path);
+	CoreParams::<NoCustom, NoCustom>::clap().gen_completions("substrate-node", *shell, &path);
```
Comment (roy-powerfinance) [inline cli/build.rs]: do we want to start using substrate-node now?

**Review by jordy25519 — APPROVED** [review]

File: cli/Cargo.toml [hunk 2]
```diff
@@ -7,34 +7,32 @@ edition = "2018"
...
+futures = { version = "0.3.1", features = ["compat"] }
 hex-literal = "0.2.1"
-jsonrpc-core = "13.2.0"
+jsonrpc-core = "14.0.3"
 log = "0.4.8"
 rand = "0.7.2"
-serde = { version = "1.0.101", features = [ "derive" ] }
-structopt = "0.3.7"
-tokio = "0.1.22"
+structopt = "=0.3.7"
```
Comment (aliXsed) [inline cli/Cargo.toml:17]: Shouldn't it be `structopt = "0.3.7"`?

File: cli/src/service.rs [hunk 3]
```diff
@@ -62,22 +59,18 @@ macro_rules! new_full_start {
...
+			let maintainer = sc_transaction_pool::FullBasicPoolMaintainer::new(pool.pool().clone(), client);
+			let maintainable_pool = sp_transaction_pool::MaintainableTransactionPool::new(pool, maintainer);
+			Ok(maintainable_pool)
 		})?
 		.with_import_queue(|_config, client, mut select_chain, _transaction_pool| {
 			let select_chain = select_chain
 				.take()
 				.ok_or_else(|| sc_service::Error::SelectChainRequired)?;
-			let (grandpa_block_import, grandpa_link) =
-				sc_finality_grandpa::block_import::<_, _, _, cennznet_runtime::RuntimeApi, _, _>(
```
Comment (aliXsed) [inline cli/src/service.rs:76]: Do we really want to remove cennznet_runtime::RuntimeApi? which is specific to cennznet?

File: cli/src/service.rs [hunk 4]
```diff
@@ -49,7 +47,6 @@ construct_simple_protocol! {
 ///
 /// Use this macro if you don't actually need the full service, but just the builder in order to
 /// be able to perform chain operations.
-#[macro_export]
```
Comment (aliXsed) [inline cli/src/service.rs:52]: This seems to be necessary

**Review by cowboy-bebug — APPROVED** [review]

**Comment by cowboy-bebug** [comment]
I'll merge this one!

File: cli/Cargo.toml (hunk 2 above)
Comment (cowboy-bebug) [reply cli/Cargo.toml:17]: We want to use the same version as in plug-rc2
https://github.com/plugblockchain/plug-blockchain/blob/1.0.0-rc2/bin/node/cli/Cargo.toml#L33

File: cli/src/service.rs (hunk 3 above)
Comment (cowboy-bebug) [reply cli/src/service.rs:76]: I think it's able to infer the type now

File: cli/src/service.rs (hunk 4 above)
Comment (cowboy-bebug) [reply cli/src/service.rs:52]: Even without `#[macro_export]`, `new_full_start!` is in scope of `cli/src/cli.rs` for some reason. I think that's why parity dropped the attribute

**Pull request merged**